### DISCUSSION TITLE
Photoprism startup Env/variable for RAM

### DISCUSSION THREAD

**Discussion by pbecks1963** [photoprism/photoprism#4478] [Q&A]
I use the photoprism plugin in Openmediavault. Photoprism is very slow when it is processing new images (6000+). How can i speed this up? 

I can pass startup variables to it (the plugin). My server has 32 Gb RAM. ( HP prodesk 600 G5, (CPU) AMD Ryzen 5 PRO 2400G with Radeon Vega Graphics.)

**Comment by srett** [comment]
I don't think there is a RAM limit, or that having more would speed things up. It's more likely to be CPU bound, or I/O if your disk is slow. Try increasing the amount of worker threads and see if it improves things. No idea how/if passing startup variables via OMV is different, [but see the docs](https://docs.photoprism.app/getting-started/config-options/) for the official way.

You probably don't want to go higher than twice the number of CPU cores, at some point you might see adverse effects. Check CPU load while indexing is running. If all cores are pegged there's no way to make it any faster.
